lib/sensu/transport/snssqs.rb

```diff
@@ -173,7 +184,19 @@ def receive_messages
             wait_time_seconds: @settings[:wait_time_seconds],
             max_number_of_messages: @settings[:max_number_of_messages],
           )
-          resp.messages
+          resp.messages.map do |msg|
+            if !msg.key? 'message_attributes'
+              tmp_body = JSON.parse msg.body
+              msg.body = tmp_body['Message']
+              msg.message_attributes = {}
+              tmp_body['MessageAttributes'].each do |name, value|
+                msg.message_attributes[name] = Aws::SQS::Types::MessageAttributeValue.new
+                msg.message_attributes[name].string_value = value['Value']
+                msg.message_attributes[name].data_type = 'String'
+              end
+            end
+            msg
```

Comment:
A message containing the message_attributes directly is a raw message and can be handled the same way it is now.